build.gradle

```diff
@@ -647,20 +647,43 @@ project(':iceberg-gcp') {
     implementation project(':iceberg-common')
     implementation project(':iceberg-core')
 
+    implementation("com.google.apis:google-api-services-bigquery:v2-rev20240602-2.0.0")
```

Comment:
Should this be in a different module? The GCP support is largely for storage right now and I'm not sure that we would want to add unexpected components. Maybe we should have an `iceberg-bigquery` module instead?

Reply:
Hmm..we could do it but it would break the pattern similar to how we have the project `:iceberg-hive-metastore` with everything Hive in there, both related to storage and metadata.
This project is also named :iceberg-gcp and BigQuery is part of GCP. So we could have two projects `iceberg-bigquery` and `iceberg-gcs` or so if you want, but I think it's honestly fine, this is just an API client library and I think it belongs here as the BigQuery code does use a fair amount of these dependencies anyway.

Reply:
I definitely prefer `iceberg-bigquery`. One of the mistakes we made with AWS was having a single giant module that ended up with many more dependencies than necessary for most people.

That also separates this out so that we don't inadvertently add all of the transitive dependencies from this new dependency into our runtime Jars without validating licenses.

Reply:
Sure. Done. Kept `iceberg-gcp` organization as-is for now but let me know if you want to split things differently or rename the existing `iceberg-gcp` project and its bundle.